openssl/src/pkey.rs

```diff
@@ -289,6 +289,21 @@ where
         private_key_to_der,
         ffi::i2d_PrivateKey
     }
+
+    pub fn private_key_to_der_pkcs8(&self) -> Result<Vec<u8>, ::error::ErrorStack> {
+        unsafe {
+            let p8inf = cvt_p(ffi::EVP_PKEY2PKCS8(self.as_ptr()))?;
```

Comment:
This library typically tries to map individual OpenSSL functions to individual Rust methods, so this would be split into a method that returns a `Pkcs8PrivKeyInfo`, and a method on that to turn that into DER.